app/jobs/maestrano/connector/rails/synchronization_job.rb

```diff
@@ -38,7 +38,7 @@ def perform(organization_id, opts = {})
         # We also do batched sync as the first one can be quite huge
         if last_synchronization.nil?
           ConnectorLogger.log('info', organization, 'First synchronization ever. Doing two half syncs to allow smart merging to work its magic.')
-          organization.synchronized_entities.select { |k, v| v }.keys.each do |entity|
+          organization.synchronized_entities.keys.each do |entity|
```

Comment:
You could left here a check to skip the entire sync for one entity if both can_push_to_external and can_push_to_connec are false.
Not sure (I'm a bit rusty), but I think that as it is, the 'get' part of the sync would still run, which may lead to some useless api calls and id maps.

Also, if can_push_to_external is false, the get_connec_entities can also be skipped to save some calls (and vice-versa)